hotifx: Fix concurrent worker updates

DancingAxolotl opened the pull request:
## Background
We had lots of strange issues recently with workers going out of sync. The most likely cause of these issues is workers running concurrently. This PR adds DB locking to prevent multiple workers from running at the same time.

## Changes
 - EthUpdater:
   -- exceptions that happen during event processing now get logged properly and block worker execution
   -- before running any operations (run or resync) the worker attempts to acquire a lock for a row in `resolution_worker_status` table with `location == worker.blockchain`
      -- if the worker cannot acquire the lock, an error is thrown and worker execution is stopped. The worker will retry on the next `FETCH_INTERVAL`
      -- if the lock is aquired, the worker proceeds to sync events
   -- worker db transactions were reworked to support locking before running any other queries

## To Do

- [ ] Fix tests

## Code Review

This Pull Request was thoroughly reviewed and meets all Code Review Standards pertaining to:

- [ ] **Implementation** - satisfied acceptance criteria
- [ ] **Communication** - notified engineers/stakeholders about impactful changes
- [ ] **Error handling** - handled, logged & alerted on errors
- [ ] **Documentation** - wrote readable code & commits, documented unintuitive code
- [ ] **Testing** - unit & E2E test coverage, tested in staging, DB migrations backwards compatible
- [ ] **Security** - sanitized inputs, vetted dependencies, validated/secured API requests, no committed secrets
- [ ] **Performance** - optimized expensive algorithms & database queries

Please select all **applied** standards relevant to this PR.
## Confirmation

- [ ] **Assignee Confirmation** - I (the author) have filled out the template above
- [ ] **Reviewer Confirmation** - I (a/the reviewer) confirm 1) the author has filled out the template and checked the box that says **Assignee Confirmation** 2) I reviewed the code and selected all applicable items in the code review section.

crazeekennee commented on src/models/WorkerStatus.ts:
```diff
@@ -85,4 +94,16 @@ export default class WorkerStatus extends Model {
+    manager: EntityManager,
+  ): Promise<void> {
+    await manager.query(
+      `LOCK TABLE resolution_worker_status IN ROW EXCLUSIVE MODE`,
```
Just so I'm clear, we're making the table only readable/editable by this one process ?

DancingAxolotl replied:
Not the whole table, but the specific row in the table. Other workers may lock other rows (e.g. ETH and MATIC can still run in parallel).

DancingAxolotl commented:
Some new updates:
 - Fixed unit tests
   - Also fixed the flaky unit tests (hopefully)
 - Unified entity manager repositories in `EthUpdater`, cleaned up queries in transactions
 - Added a validation repository for entities which need a DB query to validate themselves, so they can be validated correctly in a transaction

crazeekennee reviewed (approved):
LGTM
